Description: When a screen comes back online, the windows need to be told
 about it
Author: Ralf Jung <post@ralfj.de>
Forwarded: b0b08cc0e4e38504d6b833702f7477aee4e2a192
Last-Update: 2015-09-07

---
 src/plugins/platforms/xcb/qxcbconnection.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/src/plugins/platforms/xcb/qxcbconnection.cpp
+++ b/src/plugins/platforms/xcb/qxcbconnection.cpp
@@ -252,6 +252,14 @@ void QXcbConnection::updateScreens(const xcb_randr_notify_event_t *event)
                         otherScreen->addVirtualSibling(screen);
                 m_screens << screen;
                 QXcbIntegration::instance()->screenAdded(screen, screen->isPrimary());
+
+                // Windows which had null screens have already had expose events by now.
+                // They need to be told the screen is back, it's OK to render.
+                foreach (QWindow *window, QGuiApplication::topLevelWindows()) {
+                    QXcbWindow *xcbWin = static_cast<QXcbWindow*>(window->handle());
+                    if (xcbWin)
+                        xcbWin->maybeSetScreen(screen);
+                }
             }
             // else ignore disabled screens
         } else if (screen) {
-- 
2.4.6

